lib/index.ts

```diff
@@ -246,6 +265,91 @@ class ProgressReporter {
 		};
 	}
 
+	buildProgress(onProgress: ProgressCallback): ProgressCallback {
+		const progressRenderer = this.renderProgress;
+		let lastStep = 1;
+		let step = 0;
+		let total = 1;
+		let lastPercentage = 0;
+
+		let progressCallback: ProgressCallback = () => void 0;
+
+		return (evt: any) => {
+			let stream;
+			let id;
+			let status;
+			try {
+				({ id, stream, status } = evt);
+				if (id == null) {
+					id = '';
+				}
+				if (status == null) {
+					status = '';
+				}
+				if (stream == null) {
+					stream = '';
+				}
```

Comment:
NIT / fewer lines of code:
```suggestion
				id ||= '';
				status ||= '';
				stream ||= '';
```